### PR TITLE
fix(chore): added if condition for createall db call

### LINKED ISSUE #1124: createAll method needs array length check
**Describe the bug**
Since the latest loopback release 
if an empty array is passed to createAll method it throws an error, so a safety check is needed.
**To Reproduce**
1. Try passing an empty array to the createAll method 
2. Check error

**Expected behavior**
This should be handled at our end.

### PR DESCRIPTION
**Pull request by yeshamavani**
Breaking Change 
if the array is not passed error will be thrown 
gh-1124

## Description

Added the if condition to check the entity length before calling the repo.createAll call
to avoid error.

Fixes #1124

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Intermediate change (work in progress)

## Checklist:

- [ ] Performed a self-review of my own code
- [ ] npm test passes on your machine
- [ ] New tests added or existing tests modified to cover all changes
- [ ] Code conforms with the style guide
- [ ] API Documentation in code was updated
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: packages/core/src/repositories/default-user-modify-crud.repository.base.ts [hunk 1]
```diff
@@ -58,6 +58,9 @@ export class DefaultUserModifyCrudRepository<
       entity.createdBy = uid ?? '';
       entity.modifiedBy = uid ?? '';
     });
+    if (entities.length <= 0) {
```
Comment (akshatdubeysf) [inline packages/core/src/repositories/default-user-modify-crud.repository.base.ts:61]: Why not just check entities.length

File: services/chat-service/src/controllers/attachment-file.controller.ts [hunk 2]
```diff
@@ -96,6 +97,9 @@ export class AttachmentFileController {
     attachFileDto.attachmentFiles.forEach(messageFile => {
       attachFiles.push(new AttachmentFile(messageFile));
     });
+    if (attachFiles.length <= 0) {
```
Comment (akshatdubeysf) [inline services/chat-service/src/controllers/attachment-file.controller.ts:100]: If is at repo level, why is it required here?

**Review by akshatdubeysf — CHANGES_REQUESTED** [review]

File: services/chat-service/src/controllers/attachment-file.controller.ts (hunk 2 above)
Comment (yeshamavani) [reply services/chat-service/src/controllers/attachment-file.controller.ts:100]: will remove this

File: packages/core/src/repositories/default-transactional-user-modify-repository.base.ts [hunk 3]
```diff
@@ -57,6 +57,9 @@ export abstract class DefaultTransactionalUserModifyRepository<
       entity.createdBy = uid ?? '';
       entity.modifiedBy = uid ?? '';
     });
+    if (!entities.length) {
```
Comment (akshatdubeysf) [inline packages/core/src/repositories/default-transactional-user-modify-repository.base.ts:60]: This should be breaking change

**Review by akshatdubeysf — CHANGES_REQUESTED** [review]

**Comment by yeshamavani** [comment]
This will be released when there is a breaking change

**Comment by yeshamavani** [comment]
Raised another PR with same changes to avoid merge conflicts - https://github.com/sourcefuse/loopback4-microservice-catalog/pull/1881

**Pull request closed without merging**